javav2/usecases/topics_and_queues/src/main/java/com/example/sns/SNSWorkflow.java

```diff
@@ -0,0 +1,669 @@
+
+
+        } catch (SnsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+    }
+
+    // Subscribe to the SQS queue.
+    public static String subQueue(SnsClient snsClient, String topicArn, String queueArn, List<String> filterList) {
+        try {
+            SubscribeRequest request;
+            if (filterList.isEmpty()) {
+                // No filter subscription is added.
+                request = SubscribeRequest.builder()
+                    .protocol("sqs")
+                    .endpoint(queueArn)
+                    .returnSubscriptionArn(true)
+                    .topicArn(topicArn)
+                    .build();
+
+                SubscribeResponse result = snsClient.subscribe(request);
+                System.out.println("The queue "+queueArn +" has been subscribed to the topic "+topicArn + "\n" +
+                    "with the subscription ARN "+ result.subscriptionArn());
+                return result.subscriptionArn();
+            } else {
+                request = SubscribeRequest.builder()
+                    .protocol("sqs")
+                    .endpoint(queueArn)
+                    .returnSubscriptionArn(true)
+                    .topicArn(topicArn)
+                    .build();
+
+
+                SubscribeResponse result = snsClient.subscribe(request);
+                System.out.println("The queue "+queueArn +" has been subscribed to the topic "+topicArn + "\n" +
+                    "with the subscription ARN "+ result.subscriptionArn());
+
+                String attributeName = "FilterPolicy";
+                Gson gson = new Gson();
+                String jsonString = "{\"tone\": []}";
+                JsonObject jsonObject = gson.fromJson(jsonString, JsonObject.class);
+                JsonArray toneArray = jsonObject.getAsJsonArray("tone");
+                for (String value : filterList) {
+                    toneArray.add(new JsonPrimitive(value));
+                }
+
+                String updatedJsonString = gson.toJson(jsonObject);
+                System.out.println(updatedJsonString);
+                SetSubscriptionAttributesRequest attRequest = SetSubscriptionAttributesRequest.builder()
+                    .subscriptionArn(result.subscriptionArn())
+                    .attributeName(attributeName)
+                    .attributeValue(updatedJsonString)
+                    .build();
+
+                snsClient.setSubscriptionAttributes(attRequest);
+                return result.subscriptionArn();
+            }
+
+        } catch (SnsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+        return "";
+    }
+
+    // snippet-start:[sqs.java2.set_attributes.main]
+    // Attach a policy to the queue.
+    public static void setQueueAttr(SqsClient sqsClient, String queueUrl, String policy) {
```

Comment:
There is an SOS sqs_SetQueueAttributes. I did not see an existing Java entry. You will need to run WriteMe on sqs if you add this.